### PR TITLE
remove unnecessary __future__ imports

### PR DESCRIPTION
**Pull request by chammika-become**
This PR addresses Issue #312
- Only perform checks
- Auto fix to be attempted in a later PR

### CODE REVIEW

**Comment by charliermarsh** [comment]
Dang this is great! Thanks for the PR! Will review tonight or tomorrow morning.

File: src/pyupgrade/plugins/unnecessary_future_import.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+pub fn unnecessary_future_import(checker: &mut Checker, stmt: &Stmt, name: &str) {
+    if let Some(check) = checks::unnecessary_future_import(
+        checker.settings.target_version,
+        name,
+        Range::from_located(stmt),
+    ) {
+        // TODO - remove unnecessary import
+        // if checker.patch() {
+        // }
```
Comment (charliermarsh) [inline src/pyupgrade/plugins/unnecessary_future_import.rs:15]: Mind removing these three lines for now?

**Comment by charliermarsh** [comment]
Looks great! Happy to merge once minor comments are addressed.

File: src/pyupgrade/checks.rs [hunk 2]
```diff
@@ -4,6 +4,30 @@ use crate::ast::helpers;
 use crate::ast::types::{Binding, BindingKind, Range, Scope, ScopeKind};
 use crate::checks::{Check, CheckKind};
 use crate::pyupgrade::types::Primitive;
+use crate::settings::types::PythonVersion;
+
+pub const PY3_PLUS_REMOVE_FUTURES: &[&str] = &[
```
[the remark this answers is not in the document]
Comment (chammika-become) [reply src/pyupgrade/checks.rs]: Thanks for the catch It should be `PY33_`. 
I was modeling the remove feature after `pyupgrade` which support version 3+, 3,7, 3,10 etc. Then I realized `ruff` starts with `Py33`.

**Pull request merged**